app/persona/api/authentication/AuthenticationServiceImpl.scala

```diff
@@ -0,0 +1,38 @@
+package persona.api.authentication
+
+
+import java.util.UUID
+
+import com.google.inject.Inject
+import com.mohiva.play.silhouette.api.LoginInfo
+import com.mohiva.play.silhouette.impl.providers.CommonSocialProfile
+import persona.model.authentication.User
+import persona.model.dao.authentication.UserDAO
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class AuthenticationServiceImpl @Inject()(userDAO: UserDAO) extends AuthenticationService {
+
+  override def save(user: User): Future[User] = userDAO.save(user)
+
+  override def save(profile: CommonSocialProfile): Future[User] = {
+    userDAO.find(profile.loginInfo).flatMap {
+      case Some(user) =>
```

Comment:
What's the purpose of saving someone that's already in our database?  Pretty sure I'm just not understanding what CommonSocialProfile does, not that the code is wrong